docs/src/main/asciidoc/ansible.adoc

```diff
@@ -0,0 +1,213 @@
+skipping: [localhost]
+
+TASK [Deploy webapp on target.] ************************************************
+
+TASK [middleware_automation.quarkus.quarkus : Ensure required parameters are provided.] ***
+ok: [localhost]
+
+TASK [middleware_automation.quarkus.quarkus : Ensure required OpenJDK is installed on target.] ***
+skipping: [localhost]
+
+TASK [middleware_automation.quarkus.quarkus : Ensure Quarkus system group exists on target system] ***
+ok: [localhost]
+
+TASK [middleware_automation.quarkus.quarkus : Ensure Quarkus user exists on target system.] ***
+ok: [localhost]
+
+TASK [middleware_automation.quarkus.quarkus : Ensure deployement directory exits: /opt/quarkus_deploy.] ***
+ok: [localhost]
+
+TASK [middleware_automation.quarkus.quarkus : Set Quarkus app source dir (if not defined).] ***
+ok: [localhost]
+
+TASK [middleware_automation.quarkus.quarkus : Deploy application from  to target system] ***
+ok: [localhost]
+
+TASK [middleware_automation.quarkus.quarkus : Deploy Systemd configuration for Quarkus app] ***
+ok: [localhost]
+
+TASK [middleware_automation.quarkus.quarkus : Perform daemon-reload to ensure the changes are picked up] ***
+skipping: [localhost]
+
+TASK [middleware_automation.quarkus.quarkus : Ensure Quarkus app service is running.] ***
+ok: [localhost]
+
+TASK [middleware_automation.quarkus.quarkus : Ensure firewalld is available.] ***
+skipping: [localhost]
+
+TASK [middleware_automation.quarkus.quarkus : Configure firewall for 8080 ports] ***
+skipping: [localhost]
+
+PLAY RECAP *********************************************************************
+localhost              	: ok=15   changed=2	unreachable=0	failed=0	skipped=5	rescued=0	ignored=0
+…
+----
+
+The Ansible collection for Quarkus does all the `heavy lifting` here. First, it checks out the code from Github and builds the application from its sources. It also ensures the system used for this step does have the required OpenJDK installed. By default, the application is built on the localhost (the Ansible controller), but it can be performed on a remote system if needed. Once the application is built, the collection will take care of the deployment.
+
+Here again, it checks that the appropriate OpenJDK is available on the target system. Then we ensure that the required user and group exist on the target. This is recommended mostly to be able to run the Quarkus app with a regular user, rather than with the root account.
+
+With those requirements in place, the jar can be deployed on the target, along with the required configuration for the app integration into systemd as a service. Any change to the systemd configuration requires reloading its daemon, which the collection ensures will happen whenever it is needed. With all of that in place, the only thing that remains is to start the service itself, which Ansible will also take care of.
+
+=== Validate that deployment was successful
```

Comment:
Automate deployment and validate success manually seems really miserable, I suppose if this is ever going to be used seriously, users will ping readyness/liveness endpoints (probes)? Doesn't Ansible provide verification/validation that could ping these endpoints in last step?

Reply:
It's a tutorial, I'm giving the reader the tools to double check that what the stuff actually worked, it's not a deployment design. I'll rephrase that.

Pinging the endpoint is trivial with Ansible, you have a primitive for that (ansible.builtin.get_url), but I can add that too.

Reply:
I thought this paragraph was very far from real world, but you will know better how users are going to use this. Would it be too much of a trouble to add ping to the playbook and mention it in the text? But sure, please treat this as opinionated preference, nothing else!